api/cluster.go

```diff
@@ -86,8 +87,10 @@ func (s *Server) indexList(ctx *middleware.Context, req models.IndexList) {
 func (s *Server) getData(ctx *middleware.Context, request models.GetData) {
 	series, err := s.getTargetsLocal(request.Requests)
 	if err != nil {
+		// the only errors returned are from us catching panics, so we should treat them
+		// all as internalServerErrors
```

Comment:
>This is why I think it should be up to getTargetsLocal to describe the "kind" of error it's 

This is exactly what is being done now.  If an error is not an internalServerError then it needs to implement response.Error interface which ensures err.Code() can be called to get the type of error.   The code is expected to be a http statusCode as it provides a large set existing codes to use.

Any error that does not implement response.Error interface is treated as an internalServerError.

The comment here is just a comment.  It just provides some additional information that is not obvious from just looking at the code.